ServerCore/Helpers/SubmissionEvaluator.cs

```diff
@@ -0,0 +1,356 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+
+namespace ServerCore.Helpers
+{
+
+    public enum SubmissionResponseCode
+    {
+        Correct,
+        Incorrect,
+        Freeform,
+        Partial,
+        Unauthorized,
+        PuzzleNotFound,
+        PuzzleLocked,
+        EmptySubmission,
+        TeamLockedOut,
+        AlreadySolved,
+        DuplicateSubmission
+    }
+
+    public class SubmissionResponse
+    {
+        public SubmissionResponseCode ResponseCode { get; set; }
+        public Response CompleteResponse { get; set; }
+        public string FreeformResponse { get; set; }
+    }
+
+    public class SubmissionView
+    {
+        public Submission Submission { get; set; }
+        public Response Response { get; set; }
+        public string SubmitterName { get; set; }
+        public bool IsFreeform { get; set; }
+        public string FreeformReponse { get; set; }
+    }
+
+    public class SubmissionEvaluator
+    {
+        private static PuzzleStatePerTeam puzzleState;

```

Comment:
Should all of these fields be locals in EvaluateSubmission? They all look like they'd have race conditions as fields since they're statics